src/plugin.ts

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+import fs = require('fs');
+import path = require('path');
+import sizeTree = require('./size_tree');
+
+export class WebpackBundleSizeAnalyzerPlugin {
+  filepath: string;
+  statsOptions: object;
+  constructor(filepath: string = '', statsOptions: object = {}) {
+    this.filepath = filepath;
+    this.statsOptions = statsOptions || {};
+  }
+  apply(compiler: any) {
+    const that = this;
```

Comment:
I *thought* it was redundant, but I was running into issues.  Originally I didn't have this logic in there, but when I started working on this PR, I ran into some errors.  For this PR I had to convert my original plugin into typescript, and I also tried making it a class based plugin (instead of the functional prototype like the webpack docs).

I'll revisit this and see if I can fix it.